code/datums/weather/weather.dm

```diff
@@ -146,13 +150,15 @@
 		return
 	stage = WIND_DOWN_STAGE
 	update_areas()
+	addtimer(CALLBACK(src, .proc/end), end_duration)
 	for(var/num in impacted_z_levels)
-		for(var/M in GLOB.humans_by_zlevel["[num]"])
+		for(var/mob/M in GLOB.humans_by_zlevel["[num]"])
 			if(end_message)
```

Comment:
Also why typecast this

Reply:
If you mean why check mob, it is because it _somehow_ didn't work otherwise. No clue as to why, it didn't even compile.